pkg/apis/azure/helper/helper.go

```diff
@@ -101,11 +101,22 @@ func FindImageFromCloudProfile(cloudProfileConfig *api.CloudProfileConfig, image
 			}
 			for _, version := range machineImage.Versions {
 				if imageVersion == version.Version {
-					urn := version.URN
+					if (version.URN == nil && version.ID == nil) || (version.URN != nil && version.ID != nil) {
```

Comment:
Is this double validation really needed? You already validate the same in the `pkg/apis/azure/validation`.

Reply:
I was wondering the same and decided to put it in the PR to get some feedback.
So I will remove it.

Reply:
removed